src/app/auth/auth-registration/auth-registration.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute, ParamMap } from '@angular/router';
+
+import { AuthService } from '../auth.service';
+
+@Component({
+  selector: 'app-auth-registration',
+  templateUrl: './auth-registration.component.html',
+  styleUrls: ['./auth-registration.component.scss']
+})
+export class AuthRegistrationComponent implements OnInit {
+
+  password:String;
+  missingParams:boolean = true;
+  formValidationErrors:any = {
```

Comment:
Check https://angular.io/guide/reactive-forms#simple-form-validation and use the built in validator that Angular provided to do the form validation